tests/app/filter.test.js

```diff
@@ -31,3 +31,20 @@ test("scope filter updates markers", async ({ page }) => {
   await checkScope("Main Street/Special");
   await checkScope(["Transit Oriented", "City Center/Business District"]);
 });
+
+test("population filter updates markers", async ({ page }) => {
+  await loadMap(page);
+  const left = await page.$(".left-slider");
+  const right = await page.$(".right-slider");
+
+  const testSlider = async (slider, moveTo) => {
+    const before = await getNumCities(page);
+    await slider.fill(moveTo);
+    const after = await getNumCities(page);
+    expect(before > after).toBe(true);
```

Comment:
I think it's probably better to have a range of what we expect the values to be, like you did with the filter. That helps us to catch more edge cases with issues in the implementation.

Then, with that change, you can test some more edge cases like:

* Only 100 is selected
* 100, 500, and 1k are selected
* Everything but 100, 500, and 1k are selected

Having more edge cases covered does make the test slower to run. But that's faster than us accidentally breaking the functionality and not realizing!

Reply:
Still thinking about how to implement this. Like a data structure that will easily store and retrieve the differences